torch/csrc/monitor/events.cpp

```diff
@@ -0,0 +1,61 @@
+#include <torch/csrc/monitor/events.h>
+
+#include <shared_mutex>
+#include <sstream>
+#include <unordered_set>
+#include <vector>
+#include <algorithm>
+
+namespace torch {
+namespace monitor {
+
+namespace {
+class EventHandlers {
```

Comment:
Since we don't have the "ticker" we may miss to publish the last batch of events/metrics. Maybe add in a destructor that calls `logEvent()` for the final time or add an explicit `close()` method that the user has to call at the end of their program for now.

Reply:
done